### PR TITLE
Code cleanup and development

### PR DESCRIPTION
**Pull request by pordaniel01**
Some codesmells have been fixed. hdf5 components added. Function input checks have been added.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,6 +20,8 @@ if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
     include(CTest)
 endif()
 
+message($)
```
Comment (olaszsoma) [inline CMakeLists.txt]: This message is unnecessary.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -6,4 +6,17 @@ set(SOURCE_FILES
...
+if(HDF5_FOUND)
+	message("-- HDF5 FOUND")
+	set(SOURCE_FILES 
+			${SOURCE_FILES}
+			hdf5export.cpp)
+else()
+	message("-- HDF5 NOT FOUND")
+endif()
+
+message(${SOURCE_FILES})
```
Comment (olaszsoma) [inline src/CMakeLists.txt]: Delete this message.

File: src/dreicer.cpp [hunk 3]
```diff
@@ -52,14 +65,28 @@ double calculate_lambda(double alpha){
 }
 
 double calculate_gamma(double effective_charge, double alpha){
+
+	if(effective_charge < effective_charge_lower_boundary)
+		throw "Effective charge higher than 1";
```
Comment (olaszsoma) [inline src/dreicer.cpp]: Wrong message, fix it.

File: src/dreicer.cpp [hunk 4]
```diff
@@ -52,14 +65,28 @@ double calculate_lambda(double alpha){
...
+
+	if(effective_charge < effective_charge_lower_boundary)
+		throw "Effective charge higher than 1";
+
 	return sqrt((1.0+effective_charge) * alpha*alpha/8.0/(alpha-1.0)) * (ITM_PI/2.0-asin(1.0-2.0/alpha));
 }
 double calculate_h(double alpha, double effective_charge){
+
+	if(effective_charge < effective_charge_lower_boundary)
+		throw "Effective charge higher than 1";
```
Comment (olaszsoma) [inline src/dreicer.cpp]: Wrong message, fix it.

**Review by olaszsoma — CHANGES_REQUESTED** [review]
Fix the messages commented on. It might be cleaner if the check functions were put in separate functions instead of adding if statements in every function. This should be discussed with Gergo.

**Review by olaszsoma — APPROVED** [review]
Looks good, Gergo should review it.

File: include/constants.h [hunk 5]
```diff
@@ -57,6 +57,26 @@
...
     const double ITM_INVALID_FLOAT = -9.0e40;
 
+    //parameter boundaries
+    const double electron_density_lower_boundary = 0;
+
+    const double electron_temperature_lower_boundary = 0;
+
+    const double runaway_density_lower_boundary = 0;
+
+    const double runaway_electron_lower_boundary = 0;
```
Comment (gergopokol) [inline include/constants.h]: This might not be needed.

Comment (pordaniel01) [inline include/constants.h]: This should be removed

File: src/hdf5export.cpp [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+
+int init_hdf5_file(H5std_string FILE_NAME, int cols, string* dataset_name_list, int dataset_name_length){
+	try{
+		Exception::dontPrint();
+
+		H5File file;
+		file.openFile(FILE_NAME, H5F_ACC_RDWR);
+		file.close();
+		cout << "  [Runaway Fluid] \tHDF5 output file: "  << FILE_NAME << endl;
```
Comment (gergopokol) [inline src/hdf5export.cpp]: `Runaway Fluid` to be renamed to `Runaway Physics` everywhere.

File: test/tests.cpp [hunk 7]
```diff
@@ -69,32 +70,81 @@ const double reference_runaway_electron_density_after66 = 1.66e19;
 const double reference_runaway_electron_density_after67 = 1.31e20;
 double rate_values[4] = {0,0,0,0};
 
-//std::string module_off_setting;
-module_struct modules_off; //= {module_off_setting, false, module_off_setting, false, false, module_off_setting}
+cell CELL = {reference_rho_tor_norm, reference_ne, reference_te, reference_Zeff_1,reference_electric_field_1_runafluid, reference_magnetic_field, reference_runaway_electron_density_before};
```
Comment (gergopokol) [inline test/tests.cpp]: `reference_electric_field_1_runafluid` to be renamed to `reference_electric_field_2`. All references to runafluid are to be removed.

**Review by gergopokol — CHANGES_REQUESTED** [review]
Minor changes requested in comments. Plus #49 is to be finished.

**Review by gergopokol — APPROVED** [review]

**Pull request merged**